src/player.js

```diff
@@ -1838,7 +1841,10 @@ export default class Player extends FakeEventTarget {
       });
       this._eventManager.listen(this._engine, CustomEventType.TEXT_TRACK_CHANGED, (event: FakeEvent) => this._onTextTrackChanged(event));
       this._eventManager.listen(this._engine, CustomEventType.TRACKS_CHANGED, (event: FakeEvent) => this._onTracksChanged(event));
-      this._eventManager.listen(this._engine, CustomEventType.TEXT_CUE_CHANGED, (event: FakeEvent) => this._onCueChange(event));
+      this._eventManager.listen(this._engine, CustomEventType.TEXT_CUE_CHANGED, (event: FakeEvent) => {
+        this._onCueChange(event);
+        this.dispatchEvent(event);
```

Comment:
`TEXT_CUE_CHANGED` triggered by the external caption handler so need to consider it as well.
better to dispatch to from `_onCueChange` so it will be triggered for both.

Reply:
i added in the engine

Reply:
I moved it there, but I'm not sure about the timing, whether it should be triggered after the  _onCueChange logic as in _onTextTrackChanged or before (immediately upon the entering the function )